app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,43 @@
+class PaymentsController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  load_resource :conference, find_by: :short_title
+  authorize_resource :conference_registrations, class: Registration
+
+  def index
+    @payments = current_user.payments
+  end
+
+  def new
+    @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
+  end
+
+  def create
+    @payment = Payment.new(payment_params)
+    @total_amount_to_pay = Ticket.total_price(@conference, current_user, paid: false)
+
+    if @payment.purchase && @payment.save
+      update_purchased_ticket_purchases
+      redirect_to conference_conference_registration_path(@conference.short_title), flash: { success: 'Thanks! You have purchased your tickets successfully.' }
+    else
+      render 'new'
+    end
+  end
+
+  private
+
+  def update_purchased_ticket_purchases
+    paid_ticket_purchases = current_user.ticket_purchases.by_conference(@conference).unpaid
```

Comment:
For that to work though, you have to add `belongs_to :payment` to TicketPurchase.